test/trusted_resources_test.go

```diff
@@ -0,0 +1,290 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"github.com/tektoncd/pipeline/pkg/pod"
+	"github.com/tektoncd/pipeline/test/parse"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/system"
+	knativetest "knative.dev/pkg/test"
+	"knative.dev/pkg/test/helpers"
+)
+
+var (
+	neededFeatureFlags = map[string]string{
+		// Make sure this is running under alpha integration tests
+		"enable-api-fields": "alpha",
+	}
+	privKey  = "trustedresources-keys/cosign.key"
+	pubKey   = "trustedresources-keys/cosign.pub"
+	password = "1234"
+)
+
+func init() {
+	// This is the password for test/trustedresources-keys/cosign.key, only used in this test.
+	os.Setenv("PRIVATE_PASSWORD", password)
+}
+
+func TestTrustedResourcesVerify_Success(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	c, namespace, secretName, signer := setupResourceVerificationConfig(ctx, t, requireAnyGate(neededFeatureFlags))
+	knativetest.CleanupOnInterrupt(func() { removeResourceVerificationConfig(ctx, t, c, namespace, secretName) }, t.Logf)
+	defer removeResourceVerificationConfig(ctx, t, c, namespace, secretName)
```

Comment:
nice!! :D :D :D thanks for all the back and forth on this

Reply:
Thanks a lot!!!